buildSrc/subprojects/kotlin-dsl/src/main/kotlin/codegen/GenerateKotlinDslPluginsExtensions.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package codegen
+
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+
+import java.io.File
+
+
+@Suppress("unused")
+abstract class GenerateKotlinDslPluginsExtensions : CodeGenerationTask() {
+
+    @get:Input
+    abstract val kotlinDslPluginsVersion: Property<Any>
```

Comment:
:o: I would make this a `Property<String>` for explicitness

Reply:
That's what I had initially but it forced the use site to cast `project.version` to `String`.  I thought the additional noise was unnecessary for the task doesn't really do anything to the value other than splice it into a string template.